src/gafaelfawr/providers/github.py

```diff
@@ -300,3 +304,55 @@ async def _get_user_info(self, token: str) -> GitHubUserInfo:
             email=email,
             teams=teams,
         )
+
+    async def _get_user_teams_data(self, token: str) -> List[Dict[str, Any]]:
+        """Retrieve team membership for a user from GitHub.
+
+        Parameters
+        ----------
+        token : `str`
+            The token for that user.
+
+        Returns
+        -------
+        team_data : List[Dict[`str`, Any]]
+            Team information for that user from GitHub in GitHub's JSON
+            format.
+
+        Raises
+        ------
+        gafaelfawr.exceptions.GitHubException
+            The next URL from a Link header didn't point to the teams API URL.
+        httpx.HTTPError
+            An error occurred trying to talk to GitHub.
+        """
+        self._logger.debug("Fetching user team data from %s", self._TEAMS_URL)
+        r = await self._http_client.get(
+            self._TEAMS_URL,
+            headers={"Authorization": f"token {token}"},
+        )
+        r.raise_for_status()
+        teams_data = r.json()
+
+        # If the data was paginated, there will be a Link header with a next
+        # URL.  Retrieve each page until we run out of Link headers.
+        link_data = LinkData.from_header(r.headers.get("Link"))
+        while link_data.next_url:
+            if not link_data.next_url.startswith(self._TEAMS_URL):
```

Comment:
Does this ever happen?  Or is this just very defensive programming?

Reply:
I don't want us to send our GitHub token off to some random other web site if someone manages to MITM or alter the Link header or something.